clang/runtime/dpct-rt/include/dpct/dpl_extras/algorithm.h

```diff
@@ -937,18 +966,29 @@ stable_partition(Policy &&policy, Iter1 first, Iter1 last, Iter2 mask, Pred p) {
                        std::random_access_iterator_tag>::value,
       "Iterators passed to algorithms must be random-access iterators.");
   typedef typename std::decay<Policy>::type policy_type;
-  internal::__buffer<typename std::iterator_traits<Iter1>::value_type> _tmp(
-      std::distance(first, last));
+  auto _partition_call1 =
+      internal::make_wrapped_policy<internal::partition_call1>(policy);
+  auto _copy_call1 =
+      internal::make_wrapped_policy<internal::copy_call1>(policy);
+  auto _copy_call2 =
+      internal::make_wrapped_policy<internal::copy_call2>(policy);
```

Comment:
Can we forward `policy` the last time we use it?

Reply:
fixed